pkg/utils/cluster_names_test.go

```diff
@@ -0,0 +1,168 @@
+package utils
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/submariner-io/armada/pkg/defaults"
+)
+
+const stubConfigDir = "utils.test.dir"
+
+var origDir string
+
+func TestUtils(t *testing.T) {
+	RegisterFailHandler(Fail)
+	RunSpecs(t, "Utils test suite")
+}
+
+var _ = Describe("Utils tests", func() {
+	Context("ClusterNamesFromFiles", testClusterNamesFromFiles)
+	Context("ClusterNamesOrAll", testClusterNamesOrAll)
+})
+
+func stubDirectory() {
+	origDir = defaults.KindConfigDir
+	defaults.KindConfigDir = stubConfigDir
+}
+
+func restoreDirectory() {
+	defaults.KindConfigDir = origDir
+}
+
+func createStubDirectory() {
+	Expect(os.Mkdir(stubConfigDir, 0755)).To(Succeed())
+}
+
+func deleteStubDirectory() {
+	Expect(os.RemoveAll(stubConfigDir)).To(Succeed())
+}
+
+func generateClusterFiles(expectedClusters []string) {
+	for _, cluster := range expectedClusters {
+		fileName := fmt.Sprintf("kind-config-%s.yaml", cluster)
+		_, err := os.Create(filepath.Join(stubConfigDir, fileName))
+		Expect(err).To(Succeed())
+	}
+}
+
+func testClusterNamesFromFiles() {
+	var (
+		clusters []string
+		err      error
+	)
+
+	BeforeEach(stubDirectory)
+
+	AfterEach(restoreDirectory)
+
+	JustBeforeEach(func() {
+		clusters, err = ClusterNamesFromFiles()
+	})
+
+	When("directory doesn't exist", func() {
+		It("should return an error", func() {
+			Expect(err).To(HaveOccurred())
+		})
+	})
+
+	When("directory exists", func() {
+		BeforeEach(createStubDirectory)
+
+		AfterEach(deleteStubDirectory)
+
+		When("the directory is empty", func() {
```

Comment:
```suggestion
		Context("and is empty", func() {
```

This way the output reads "when the directory exists and is empty" which I think is cleaner.